mod/research/set/self/research_page/slot_machine.haml

```diff
@@ -0,0 +1,134 @@
+                = company
+            - else
+              .dropdown
+                %a.company-color.dropdown-toggle.d-flex.flex-nowrap.no-decoration.align-items-center{href:"#", data: { toggle: "dropdown" } }
+                  .image-box.icon
+                    %span.img-helper
+                    = nest [company, :image], view: :core, size: :small
+                  = company
+                .dropdown-menu.scrollable-menu
+                  - company_list.each do |company|
+                    .dropdown-item.company-color
+                      %a.inherit-anchor.no-decoration{href: research_url(company: company)}
+                        %h5
+                          = company
+                    .dropdown-divider
+          - if project?
+            -#= next_button :company
+            -#.w-25
+            -#  = nest [company, project], view: :research_progress_bar
+      - if metric? && company?
+        .year.border-bottom.p-2.pl-3.d-flex.flex-nowrap.justify-content-between.align-items-center
+          .large-text
+            - if year_pinned?
+              .d-flex.flex-nowrap
+                .image-box.icon.bg-white
+                  %span.img-helper
+                  = fa_icon :calendar, class: "text-muted"
+                = year
+            - else
+              .dropdown
+                %a.inherit-anchor.no-decoration.dropdown-toggle.d-flex.flex-nowrap.align-items-center{ href: "#", data: { toggle: "dropdown" } }
+                  .image-box.icon.bg-white
+                    %span.img-helper
+                    = fa_icon :calendar, class: "text-muted"
+                  - if year?
+                    %strong
+                      = year
+                  - else
+                    %span.text-muted
+                      Year
+                .dropdown-menu.scrollable-menu.slab
+                  - year_list.each do |year|
+                    .dropdown-item
+                      %a.inherit-anchor.no-decoration{ href: research_url(year: year)}
+                        %h5.d-flex.flex-no-wrap.text-muted
+                          %div.mr-5
+                            %strong
+                              = year
+                          -if record_card.answers_by_year[year]
+                            %div
+                              = nest record_card.answers_by_year[year], view: :concise, hide: :year
+                          -else
+                            %i
+                              add answer
+          - if project?
+            -#.w-25
+            -#  = nest [year.to_s, project], view: :research_progress_bar
+
+      -if year && metric && company
+        .border-bottom.background-grey.p-2.pl-3
+          %h5
+            Question
+          .large-text
+            = nest [metric, :question], view: :core
+        .p-2.pl-3
+          = answer_slot
+
+
+    .col-7.border-left#research-details
+      = right_side_tabs
```

Comment:
record for longest WikiRate haml file = crushed.

I'm not sure there's a good reason to break it up, though.

Reply:
there is a lot of repetition but also a lot of little differences